test/BoardTests.cpp

```diff
@@ -70,10 +70,147 @@ TEST_F(BoardTests, checkBoardArgumentBoard_WithGetBoardLinesFunction)
     Lines vec2 = {vec, vec};
 
     Board board(3, 3, vec2, vec2);
-    ASSERT_EQ(board.getBoardLines().at(1)[1],BoardFields::UNKNOWN);
-    ASSERT_EQ(board.getBoardLines().at(1)[0],BoardFields::UNKNOWN);
-    ASSERT_EQ(board.getBoardLines().at(0)[1],BoardFields::UNKNOWN);
-    ASSERT_EQ(board.getBoardLines().at(0)[0],BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1)[1], BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1)[0], BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0)[1], BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0)[0], BoardFields::UNKNOWN);
+}
+
+TEST_F(BoardTests, checkBoardSetField)
+{
+    Board board(3, 2, {}, {});
+    board.setField(0, 0, BoardFields::BLACK);
+    board.setField(1, 0, BoardFields::WHITE);
+
+    ASSERT_EQ(board.getBoardLines().at(0).at(0), BoardFields::BLACK);
+    ASSERT_EQ(board.getBoardLines().at(1).at(1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(2).at(0), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(0).at(1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1).at(0), BoardFields::WHITE);
+    ASSERT_EQ(board.getBoardLines().at(2).at(1), BoardFields::UNKNOWN);
+
+    board.setField(0, 0, BoardFields::UNKNOWN);
+    board.setField(2, 0, BoardFields::BLACK);
+    board.setField(1, 0, BoardFields::UNKNOWN);
+    board.setField(2, 1, BoardFields::WHITE);
+
+    ASSERT_EQ(board.getBoardLines().at(0).at(0), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1).at(1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(2).at(0), BoardFields::BLACK);
+    ASSERT_EQ(board.getBoardLines().at(0).at(1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(1).at(0), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getBoardLines().at(2).at(1), BoardFields::WHITE);
+}
+
+TEST_F(BoardTests, checkBoardGetField)
+{
+    Board board(3, 2, {}, {});
+    board.setField(0, 0, BoardFields::BLACK);
+    board.setField(0, 1, BoardFields::WHITE);
+
+    ASSERT_EQ(board.getField(1, 1), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getField(1, 0), BoardFields::UNKNOWN);
+    ASSERT_EQ(board.getField(0, 1), BoardFields::WHITE);
+    ASSERT_EQ(board.getField(0, 0), BoardFields::BLACK);
+}
+
+TEST_F(BoardTests, checkBoardSetFieldException)
+{
+    Board board(2, 3, {}, {});
```

Comment:
You should move board to BoardTests class, so that you don't need to put it in every test. You can name it eg. board23 if you want to have more boards.

Reply:
not needed anymore now